### PR TITLE
Fix Visual Studio error due to alternate syntactic form of noreturn

### PR DESCRIPTION
**Pull request by daanzu**

### CODE REVIEW

**Comment by danpovey** [comment]
@daanzu  what version of Visual Studio did you get the error on?  Looks like this change may break some versions of gcc.

**Comment by daanzu** [comment]
@danpovey Visual Studio 2017 (15.9.4)

File: src/base/kaldi-error.cc [hunk 1]
```diff
@@ -221,7 +221,7 @@ FatalMessageLogger::FatalMessageLogger(LogMessageEnvelope::Severity severity,
   }
 }
 
-[[ noreturn ]] FatalMessageLogger::~FatalMessageLogger() noexcept(false) {
+FatalMessageLogger::~FatalMessageLogger [[ noreturn ]] () noexcept(false) {
```
Comment (danpovey) [inline src/base/kaldi-error.cc]: @daanzu would you mind making a PR to make it more like the following, so we can support both compiler versions?  Obviously test that it works first.

```
#if defined(_MSC_VER)
FatalMessageLogger::~FatalMessageLogger [[ noreturn ]] () noexcept(false) 
#else
[[noreturn]] FatalMessageLogger::~FatalMessageLogger () noexcept(false) 
#endif
{
```

Comment (daanzu) [reply src/base/kaldi-error.cc]: 👍

**Pull request merged**